src/engine/renderer/tr_shader.cpp

```diff
@@ -5992,209 +6000,192 @@ A second parameter will cause it to print in sorted order
 */
 void R_ShaderList_f()
 {
-	int      i;
-	int      count;
-	shader_t *shader;
-	const char *s = nullptr;
-
-	Log::Notice("-----------------------" );
-
-	if ( ri.Cmd_Argc() > 1 )
-	{
-		s = ri.Cmd_Argv( 1 );
-	}
-
-	count = 0;
-
-	for ( i = 0; i < tr.numShaders; i++ )
-	{
-		if ( ri.Cmd_Argc() > 2 )
-		{
-			shader = tr.sortedShaders[ i ];
-		}
-		else
-		{
-			shader = tr.shaders[ i ];
-		}
+	const char *s = ri.Cmd_Argc() > 1 ? ri.Cmd_Argv( 1 ) : nullptr;
+
+	std::map<shaderType_t, std::string> shaderTypeStrings;
+	shaderTypeStrings[shaderType_t::SHADER_2D] = "2D";
+	shaderTypeStrings[shaderType_t::SHADER_3D_DYNAMIC] = "3D_DYNAMIC";
+	shaderTypeStrings[shaderType_t::SHADER_3D_STATIC] = "3D_STATIC";
+	shaderTypeStrings[shaderType_t::SHADER_LIGHT] = "LIGHT";
+
+	std::string shaderTypeHeader = "shaderType";
+	int maxShaderTypeStringLen = shaderTypeHeader.length();
+
+	for ( std::map<shaderType_t, std::string>::iterator iter = shaderTypeStrings.begin();
+		iter !=shaderTypeStrings.end(); iter++ )
+	{
+		maxShaderTypeStringLen = std::max( maxShaderTypeStringLen, (int)iter->second.length() );
+	}
+
+	std::map<shaderSort_t, std::string> shaderSortStrings;
```

Comment:
Use `unordered_map` (hash table) rather than `map` (binary tree) unless there is a specific reason you need it.

Reply:
Done. I also did things like `using shaderSortStringMap = std::unordered_map<shaderSort_t, std::string>;` to avoid repeating `std::unordered_map<shaderSort_t, std::string>` everywhere the type was needed.